### PR TITLE
test: fixed failing missed k8s tests

### LINKED ISSUE #12867: test: Running individual tests is no longer possible
```
$ NETNEXT=true KUBEPROXY=0 K8S_NODES=3 NO_CILIUM_ON_NODE=k8s3 BUILD_NUMBER=2 ginkgo --v --focus 'K8sService.*' --focus '.*LoadBalancer.*' -- --cilium.showCommands=false --cilium.provision=false --cilium.holdEnvironment=true --cilium.testScope="k8s" --cilium.passCLIEnvironment=true
[...]
12:06:36 STEP: Installing Cilium

---
K8sServicesTest Checks service across nodes Tests NodePort BPF Tests with direct routing Tests LoadBalancer Connectivity to endpoint via LB
at /home/brb/sandbox/gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514

[Cilium cannot be installed
Expected
    <*helpers.cmdError | 0xc0004b0e20>: Unable to generate YAML (Process exited with status 1) output: cmd: helm template --validate /home/vagrant/go/src/github.com/cilium/cilium/install/kubernetes/cilium --namespace=kube-system  --set global.autoDirectNodeRoutes=true  --set global.ci.kubeCacheMutationDetector=true  --set global.tag=latest  --set global.logSystemLoad=true  --set global.k8sServicePort=6443  --set global.tunnel=disabled  --set global.bpf.preallocateMaps=true  --set global.devices='{enp0s8,enp0s3}'  --set global.k8s.requireIPv4PodCIDR=true  --set config.sessionAffinity=false  --set hubble-relay.image.tag=latest  --set global.debug.enabled=true  --set global.hubble.listenAddress=:4244  --set global.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].operator=NotIn  --set preflight.image=cilium-dev  --set global.hostFirewall=true  --set config.bpfMasquerade=true  --set global.registry=k8s1:5000/cilium  --set agent.image=cilium-dev  --set global.etcd.leaseTTL=30s  --set config.enableCnpStatusUpdates=true  --set operator.image=operator  --set global.pprof.enabled=true  --set global.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key=cilium.io/ci-node  --set global.k8sServiceHost=192.168.36.11  --set global.kubeProxyReplacement=strict  --set global.ipv4.enabled=true  --set global.hubble.enabled=true  --set global.nativeRoutingCIDR=10.0.0.0/8  --set global.ipv6.enabled=true  --set global.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].values[0]=k8s3  > cilium-162acbd5129db15a.yaml
Exitcode: 1
Err: Process exited with status 1
Stdout:

Stderr:
         Error: rendered manifests contain a resource that already exists. Unable to continue with install: existing resource conflict: namespace: kube-system, name: cilium, existing_kind: /v1, Kind=ServiceAccount, new_kind: /v1, Kind=ServiceAccount
```

The following PR https://github.com/cilium/cilium/pull/12763/files#diff-2304891712e5d0140828499fc3a19043R2305-R2354 removed the cleanup of the resources.

Removing `--validate` from `helm template` can be used as a temporary mitigation.

### PR DESCRIPTION
**Pull request by fristonio**
See individual commit messages for more details.

Fixes #12867

### CODE REVIEW

**Comment by fristonio** [comment]
test-me-please

File: install/kubernetes/cilium/charts/preflight/templates/clusterrolebinding.yaml [hunk 1]
```diff
@@ -8,5 +8,5 @@ roleRef:
   name: cilium-pre-flight
 subjects:
 - kind: ServiceAccount
-  name: cilium
+  name: cilium-pre-flight
```
Comment (aanm) [inline install/kubernetes/cilium/charts/preflight/templates/clusterrolebinding.yaml:11]: Please add in the commit message the "Why?" we need this change.

**Review by aanm — CHANGES_REQUESTED** [review]
Small nit

**Comment by fristonio** [comment]
test-focus K8sUpdates

**Comment by fristonio** [comment]
test-missed-k8s

**Comment by fristonio** [comment]
test-focus K8sServicesTest

**Review by christarazi — APPROVED** [review]
LGTM, there's a typo in one of serviceAccount commit: `serviceAccnout`

**Comment by fristonio** [comment]
test-focus K8sServicesTest

**Comment by fristonio** [comment]
test-me-please

**Review by sayboras — APPROVED** [review]
LGTM and thanks.

I can re-run gingko test with `--cilium.provision=false` now :man_dancing:

**Comment by fristonio** [comment]
test-missed-k8s

**Comment by fristonio** [comment]
test-missed-k8s

**Comment by fristonio** [comment]
test-me-please

**Comment by fristonio** [comment]
test-focus K8sServicesTest

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -3268,7 +3268,7 @@ func (kub *Kubectl) GatherLogs(ctx context.Context) {
 		"kubectl get replicationcontroller --all-namespaces -o json":         "replicationcontroller.txt",
```
Comment (aanm) [inline test/helpers/kubectl.go:3261]: In the commit message, it should be:
```
Fixes: aeefd5862ae4 ("test/helpers: dump rbac information on k8s test failure")
Signed-off-by: Deepesh Pathak <deepshpathak@gmail.com>
```

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by fristonio** [comment]
test-focus K8sUpdates

**Review by aanm — APPROVED** [review]

**Comment by fristonio** [comment]
test-focus K8sServicesTest

**Comment by fristonio** [comment]
test-focus K8sUpdates

**Comment by fristonio** [comment]
test-missed-k8s

**Review by tklauser — APPROVED** [review]

**Comment by fristonio** [comment]
test-me-please

**Comment by fristonio** [comment]
retest-net-next

**Comment by fristonio** [comment]
test-focus K8sServicesTest

**Review by aanm — APPROVED** [review]

**Comment by fristonio** [comment]
test-focus K8sServicesTest

**Pull request merged**